sql/hive/src/test/scala/org/apache/spark/sql/hive/PartitionProviderCompatibilitySuite.scala

```diff
@@ -81,7 +81,7 @@ class PartitionProviderCompatibilitySuite
           HiveCatalogMetrics.reset()
           assert(spark.sql("select * from test where partCol < 2").count() == 2)
           assert(HiveCatalogMetrics.METRIC_PARTITIONS_FETCHED.getCount() == 2)
-          assert(HiveCatalogMetrics.METRIC_FILES_DISCOVERED.getCount() == 2)
+          assert(HiveCatalogMetrics.METRIC_FILES_DISCOVERED.getCount() == 7)
```

Comment:
what happened here?

Reply:
all the files should be parsed once for creating file index. So it is `5 + 2`